### ISSUE TITLE
How to get information to build links for pagination

### ISSUE THREAD

**Issue by ieugen** [katharsis-project/katharsis-core#307]
Hello, 

I'm posting this partly because I need help and partly to gather information for documentation. Once it is done we can move it to official docs.

I need to build links for pagination:
````
"links": {
    "self": "http://example.com/api/v1/articles?page[number]=3&page[size]=10",
    "first": "http://example.com/api/v1/articles?page[number]=1&page[size]=10",
    "prev": "http://example.com/api/v1/articles?page[number]=2&page[size]=10",
    "next": "http://example.com/api/v1/articles?page[number]=4&page[size]=10",
    "last": "http://example.com/api/v1/articles?page[number]=13&page[size]=10"
  }
````

I'm using `@JsonApiLinLinks` annotation. I need to extract information from the request and/or the registry to know:

- what protocol, host and port I'm using (we might use another value if if we are behind proxy)
- where the resource is mounted (`api/v1/` in my case) ??
- the resource name: `articles` in the example ??
- paging information: extracted from QueryParams

**Comment by cparker92** [comment]
Katharsis should use the same parameter providers on the @JsonApiLinks method, just like it does on the findOne or findAll repo methods. E.g. for jax-rs, the @JsonApiLinks method looks like:

``` java
@JsonApiLinks
public PaginationLinksInfo getLinksInformation(Iterable<SearchHit> resources, QueryParams queryParams, ContainerRequestContext requestContext) throws MalformedURLException {
    ...
 }
```

which will include the same QueryParams and ContainerRequestContext objects that were used for the corresponding findOne/All invocation.

**Comment by meshuga** [comment]
Since v 2.4.0 it's possible to return meta and links information along with the request i.e.
```
@JsonApiResourceRepository(Task.class)
public class TaskRepository {

    @JsonApiFindAll
    public JsonApiResponse findAll() {
        return new JsonApiResponse()
            .setEntity(Collections.singletonList(new Task(1L, "John")))
            .setLinksInformation(new LinksInformation() {
                public String self = "...";
                public String first = "...";
                public String prev = "...";
                public String next = "...";
                public String last = "...";
            });
    }
}
```
I think this is the most elegant way of creating links and should be put into documentation :)

**Comment by masterspambot** [comment]
@ieugen could you update katharsis.io docs?
